File: demo/go/ceapi.go

package main

import (
	"fmt"
	"io/ioutil"
	"net/http"
	"strings"
	"crypto/hmac"
	"crypto/sha256"
	"crypto/md5"
	"encoding/hex"
)
//网站域名
var base_url    = "https://api.coinegg.im"
//公钥
var public_key  = "wb55j-jrdw8-byghj-n8b8y-zcc3y-r88z9-i7ykm"
//私钥
var private_key = md5Encry("zx6pA-hXm7/-sBg~z-nCiP4-[Dr;H-[Uk25-mSOkK")

func main(){
	//获取账户余额
	getBalance();
}
	
//获取账户余额	
func getBalance(){

	var url = base_url + "/api/v1/balance/"
	params := "key="+public_key
	params += "&signature="+getSha256Encry(params)
	result := httpRequest("POST", url, params)

	fmt.Println(result)
}

//sha 256加密
func getSha256Encry(str string) string{
	privateKeyByte := []byte(private_key)
	h := hmac.New(sha256.New, privateKeyByte)
	h.Write([]byte(str))
	return hex.EncodeToString(h.Sum(nil))
}

//md5加密
func md5Encry(text string) string {
   ctx := md5.New()
   ctx.Write([]byte(text))
   return hex.EncodeToString(ctx.Sum(nil))
}

//http请求
func httpRequest(method string, url string, params string) string {
    request, _ := http.NewRequest(method, url, strings.NewReader(params))
    request.Header.Set("Content-Type", "application/x-www-form-urlencoded; charset=UTF-8")

    response,err := http.DefaultClient.Do(request)

    if err != nil {
        fmt.Printf("http error:%v\n", err)
        return ""
    } else {
        respBody,_ := ioutil.ReadAll(response.Body)
        return string(respBody)
    }
}

File: demo/python/ceapi.py

import hmac
import hashlib
import time
import json
import requests


url         = "https://api.coinegg.im"
private_key = "zx6pA-hXm7/-sBg~z-nCiP4-[Dr;H-[Uk25-mSOkK"
public_key  = "wb55j-jrdw8-byghj-n8b8y-zcc3y-r88z9-i7ykm"

def hmac_sha256(access_key, params):
    str = params.encode("utf-8")
    md5_key = md5_sign(access_key)
    hmac_obj = hmac.new(md5_key.encode('utf-8'), str, digestmod=hashlib.sha256);

    return hmac_obj.hexdigest().lower()

def md5_sign(str):
    hl_obj = hashlib.md5()
    hl_obj.update(str.encode("utf-8"))
    return hl_obj.hexdigest().lower()

def get_signature(params):
    params= sorted(params.items(), key=lambda item:item[0])
    params_str = ""
    for i in params:
        params_str += i[0]+"="+str(i[1])+"&"

    params_str = params_str.rstrip('&')
    return hmac_sha256(private_key, params_str)

def getBalance():
    params = {
            'key': public_key,
            }

    params['signature'] = get_signature(params)
    print("signature:"+params['signature'])
    params = OrderedDict(sorted(params.items(), key=lambda item:item[0]))
    response = requests.post(url+"/api/v1/balance/", data = params)
    return response.text

print(getBalance())
